api/src/test/java/org/apache/iceberg/transforms/TestBucketing.java

```diff
@@ -165,6 +159,68 @@ public void testLong() {
         .isEqualTo(hashBytes(buffer.array()));
   }
 
+  @Test
+  public void testTimestampNanoPromotion() {
+    // Values from spec Appendix B: 32-bit Hash Requirements
+    String timestamp1 = "2017-11-16T22:31:08";
+    long expectedHash1 = -2047944441;
+    String timestamp2 = "2017-11-16T22:31:08.000001";
+    long expectedHash2 = -1207196810;
+    String timestampNs1 = "2017-11-16T22:31:08";
```

Comment:
this is the same as `timestamp1`

Reply:
This is obviated by the requested inlining.